pshtt.py

```diff
@@ -149,13 +150,17 @@ def basic_check(endpoint):
         except requests.exceptions.SSLError:
             # If it's a protocol error or other, it's not live.
             endpoint.live = False
+            logging.warn("Unexpected SSL protocol (or other) error during retry.")
             return
         except requests.exceptions.RequestException:
             endpoint.live = False
-            logging.warn("Unexpected requests exception during retry.")
+            logging.warn("Unexpected requests exception during retry. Printing error:")
+            logging.warn(utils.format_last_exception())
             return
 
         # If it was a certificate error of any kind, it's live.
+        endpoint.live = True
+
```

Comment:
Amid all the try/except blocks and conditionals, I wasn't proactively marking `endpoint.live` as `True` in one of them, which occasionally caused an endpoint to get marked as being down and for the canonical endpoint to change.